src/ucp/proto/proto_select.c

```diff
@@ -1170,8 +1172,46 @@ void ucp_proto_select_get_valid_range(
         } else {
             /* Protocol is valid, so extend range end */
             *max_length_p = max_msg_length;
+            found         = 1;
         }
 
         ++elem;
     } while (max_msg_length < SIZE_MAX);
+
+    return found;
```

Comment:
maybe remove `found` local variable and have:
```
return `*max_length_p != 0`;
```
and also it worth adding assertion that `max_msg_length > 0`

Reply:
but max_msg_length could be 0 (support only 0 byte message)